src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Duke {
+
+    private Ui ui;
+    private TaskList tasks;
+    private Storage storage;
+
+
+    /**
+     * Constructor for Duke class
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException e) {
+            ui.showFileError();
+        }
+    }
+
+    /**
+     * Method to run entire programme
+     */
+    public String getResponse(String input) throws DukeException {
+
+        try {
+            String[] commands = Parser.parse(input);
+            if (commands[0].equals("bye")) {
+                return ui.showBye();
+
+            } else if (commands[0].equals("list")) {
```

Comment:
Spacing could be more consistent in-line with the rest of the if-else statements here. You might want to consider using switch-case here to better differentiate levels of nesting as well.